src/components/structures/ContextualMenu.js

```diff
@@ -85,6 +88,26 @@ export default class ContextualMenu extends React.Component {
         });
     }
 
+    onContextMenu(e) {
+        if (this.props.closeMenu) {
+            this.props.closeMenu();
+
+            e.preventDefault();
+            const x = e.clientX;
+            const y = e.clientY;
+
+            setImmediate(() => {
+                const clickEvent = document.createEvent('MouseEvents');
+                clickEvent.initMouseEvent(
+                    'contextmenu', true, true, window, 0,
+                    0, 0, x, y, false, false,
+                    false, false, 0, null,
+                );
+                document.elementFromPoint(x, y).dispatchEvent(clickEvent);
+            });
```

Comment:
Hmm, this is pretty grim, but I guess it's the only option apart from changing this to use a listener on document to dismiss the menu rather than a giant div that covers everything. Maybe deserves an XXX?